analyzers/maven/maven.go

```diff
@@ -141,16 +141,26 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 			Command: a.Options.Command,
 		})
 		if err != nil {
+			// Because this was a custom shell command, we do not fall back to any other strategies.
 			return graph.Deps{}, err
 		}
 		imports, deps, err = maven.ParseDependencyTree(output)
 	}
 	if err != nil {
-		return graph.Deps{}, err
+		log.Warnf(
+			"Could not use Maven to determine dependencies for %q. Falling back to use manifest file.",
+			a.Module.Name)
+		mvnError := err
+		pom, err := maven.ResolveManifestFromBuildTarget(a.Module.BuildTarget)
+		if err != nil {
+			return graph.Deps{},
+				errors.Wrapf(err, "could not identify dependencies; original error: %v", mvnError)
+		}
+		a.Maven.AddCachedManifest(a.Module.BuildTarget, pom)
```

Comment:
I think manifest caching should be transparent to the consumer -- we shouldn't have to manually add the cached manifest here. Can we do it in `ResolveManifestFromBuildTarget` instead?

Reply:
Indeed every invocation of `ResolveManifestFromBuildTarget` is followed by saving the manifest object to the cache. I too thought about bundling the cacheing into `ResolveManifestFromBuildTarget` but then leaned more toward making `ResolveManifestFromBuildTarget` simpler. With your suggestion I'll go ahead and revise. :)